src/message.rs

```diff
@@ -0,0 +1,256 @@
+) -> Result<JobResult, MessageError> {
+  let source_path = job
+    .get_string_parameter(SOURCE_PATH_PARAMETER)
+    .ok_or_else(|| {
+      MessageError::ProcessingError(
+        job_result
+          .clone()
+          .with_status(JobStatus::Error)
+          .with_message(&format!(
+            "Invalid job message: missing expected '{}' parameter.",
+            SEGMENT_DURATION_PARAMETER
+          )),
+      )
+    })?;
+
+  let segment_duration = job
+    .get_integer_parameter(SEGMENT_DURATION_PARAMETER)
+    .ok_or_else(|| {
+      MessageError::ProcessingError(
+        job_result
+          .clone()
+          .with_status(JobStatus::Error)
+          .with_message(&format!(
+            "Invalid job message: missing expected '{}' parameter.",
+            SEGMENT_DURATION_PARAMETER
+          )),
+      )
+    })?;
+
+  let segment_overlap = job.get_integer_parameter(SEGMENT_OVERLAP_PARAMETER);
+
+  let media_duration = get_media_duration_in_milliseconds(&source_path).map_err(|msg| {
+    MessageError::ProcessingError(
+      job_result
+        .clone()
+        .with_status(JobStatus::Error)
+        .with_message(&msg),
+    )
+  })?;
+
+  debug!("Input media duration: {} ms", media_duration);
+
+  let segments = split_media(media_duration, segment_duration, segment_overlap).map_err(|msg| {
+    MessageError::ProcessingError(
+      job_result
+        .clone()
+        .with_status(JobStatus::Error)
+        .with_message(&msg),
+    )
+  })?;
+
+  let result = serde_json::to_string(&segments).map_err(|error| {
+    MessageError::ProcessingError(
+      job_result
+        .clone()
+        .with_status(JobStatus::Error)
+        .with_message(&format!("{}", error)),
+    )
+  })?;
+
+  Ok(
+    job_result
+      .with_status(JobStatus::Completed)
+      .with_message(&result),
+  )
+}
+
+fn get_media_duration_in_milliseconds(path: &str) -> Result<u64, String> {
+  let mut format_context = FormatContext::new(path)?;
+  format_context.open_input()?;
+
+  let duration_millisec = if let Some(duration) = format_context.get_duration() {
+    (duration * 1000.) as u64
+  } else {
+    0
+  };
+
+  format_context.close_input();
+  Ok(duration_millisec)
+}
+
+fn split_media(
+  media_duration: u64,
+  segment_duration: i64,
+  segment_overlap: Option<i64>,
+) -> Result<Vec<Segment>, String> {
+  let number_of_segments = (media_duration as f64 / segment_duration as f64).ceil() as u64;
+  let overlap = segment_overlap.unwrap_or(0) as u64;
+
+  let mut segments = Vec::with_capacity(number_of_segments as usize);
+  let mut next_start = 0;
+  let mut next_end = 0;
+
+  for _ in 0..number_of_segments {
+    next_end += segment_duration as u64 - 1;
+    if next_end >= media_duration {
+      next_end = media_duration - 1;
+    }
+
+    segments.push(Segment::new(next_start, next_end));
+
+    next_end += 1;
+    if next_end < overlap {
+      next_start = 0;
+    } else {
+      next_start = next_end - overlap;
+    }
+  }
```

Comment:
missing case of `segment_duration * numbe_of_segments > media_duration`

Reply:
No, it is supported [just above, line 123](https://github.com/media-cloud-ai/rs_media_splitter_worker/pull/1/files#diff-f619358e9c7b971d9e11c0b99f8a30a3R123)
```rust
    next_end += segment_duration as u64 - 1;
    if next_end >= media_duration {
      next_end = media_duration - 1;
    }
```